<launch>
    <arg name="config_file" default="$(find rwth_upper_body_detector)/config/config_Asus.inp" />
    <arg name="template_file" default="$(find rwth_upper_body_detector)/config/upper_temp_n.txt" />
    <arg name="queue_size" default="5" />
    <arg name="camera_namespace" default="/head_xtion" />
    <arg name="upper_body_detections" default="/upper_body_detector/detections" />
    <arg name="upper_body_annotations" default="/upper_body_detector/annotations" />
    <arg name="upper_body_bb_centres" default="/upper_body_detector/bounding_box_centres" />
    <arg name="upper_body_image" default="/upper_body_detector/image" />
    <arg name="ground_plane" default="/ground_plane" />
    <arg name="detected_persons" default="/detected_persons"/>
    <arg name="detection_id_offset" default="0"/>
    <arg name="detection_id_increment" default="1"/>

    <node pkg="rwth_upper_body_detector" type="upper_body_detector" name="upper_body_detector" output="screen" respawn="true">
        <param name="config_file" value="$(arg config_file)" type="string"/>
        <param name="template_file" value="$(arg template_file)" type="string"/>
        <param name="queue_size" value="$(arg queue_size)" type="int"/>
        <param name="camera_namespace" value="$(arg camera_namespace)" type="string"/>
        <param name="upper_body_detections" value="$(arg upper_body_detections)" type="string"/>
        <param name="upper_body_annotations" value="$(arg upper_body_annotations)" type="string"/>
        <param name="upper_body_bb_centres" value="$(arg upper_body_bb_centres)" type="string"/>
        <param name="upper_body_image" value="$(arg upper_body_image)" type="string"/>
        <param name="ground_plane" value="$(arg ground_plane)" type="string"/>
        <param name="detected_persons" value="$(arg detected_persons)" type="string"/>
        <param name="detection_id_offset" value="$(arg detection_id_offset)"/>
        <param name="detection_id_increment" value="$(arg detection_id_increment)"/>
    </node>

</launch>
